includes/deploy/class-crio-utility.php

```diff
@@ -85,4 +85,143 @@ public static function register_template_locations() {
 			}
 		}
 	}
+
+	/**
+	 * Set Custom Templates.
+	 *
+	 * This is run after all pages / posts are added to the site.
+	 * This is used to adjust the theme_mods for the custom templates and ensure that
+	 * they match the new post ids. This is necessary because the theme_mods are set
+	 * using the post id of the template before being submitted by the author plugin, which
+	 * is different thant the post id when created via inspirations.
+	 *
+	 * @param array $cph_original_ids Original CPH Id Numbers.
+	 */
+	public static function set_custom_templates( $cph_original_ids ) {
+		$template_locations = array( 'page_headers', 'page_footers', 'sticky_page_headers' );
+		$page_post_types    = array( 'global', 'pages', 'posts', 'home', 'blog', 'search' );
+
+		$theme_mods = array();
+
+		/*
+		 * The theme mods that store the various custom template settings are a combination of
+		 * the template location and the post and page type that the template is for. For example,
+		 * 'bgtfw_page_header_home_template'. Rather than list all of the combinations, we loop
+		 * through the two arrays of combination possibilities and set the theme mods.
+		 */
+		foreach ( $template_locations as $location ) {
+			foreach ( $page_post_types as $page_post_type ) {
+				$theme_mod_name = 'bgtfw_' . $location . '_' . $page_post_type . '_template';
+				$template_id = get_theme_mod( $theme_mod_name );
+				if ( isset( $cph_original_ids[ $template_id ] ) ) {
+					set_theme_mod( $theme_mod_name, $cph_original_ids[ $template_id ] );
+				}
+				$theme_mods[ $theme_mod_name ] = get_theme_mod( $theme_mod_name );
```

Comment:
What's the purpose for doing this?

Reply:
I thought I explained it pretty well in the comments